File: FilFeed/src/main/java/com/pluea/filfeed/util/NetworkUtil.java
package com.pluea.filfeed.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.Network;
import android.net.NetworkInfo;

import java.util.regex.Matcher;
import java.util.regex.Pattern;


public class NetworkUtil {

	public static boolean isWifiConnected(Context context) {
		ConnectivityManager connectivityManager = (ConnectivityManager)context.getSystemService(Context.CONNECTIVITY_SERVICE);
		NetworkInfo info = connectivityManager.getActiveNetworkInfo();
		return info.isConnected() && info.getTypeName().equals("WIFI");
	}
}

File: domain/src/main/java/com/phicdy/mycuration/domain/util/NetworkUtil.kt
package com.phicdy.mycuration.domain.util

import android.content.Context
import android.net.ConnectivityManager


object NetworkUtil {

    fun isWifiConnected(context: Context): Boolean {
        val connectivityManager = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
        val info = connectivityManager.activeNetworkInfo
        return info != null && info.isConnected && info.typeName == "WIFI"
    }
}
